system-contracts/contracts/ContractDeployer.sol

```diff
@@ -22,6 +22,12 @@ import {Unauthorized, InvalidNonceOrderingChange, ValueMismatch, EmptyBytes32, N
  * do not need to be published anymore.
  */
 contract ContractDeployer is IContractDeployer, SystemContractBase {
+    /// @dev Prefix for EVM contracts hashes storage slots.
+    uint256 private constant EVM_HASHES_PREFIX = 1 << 254;
+    /// @dev keccak256("ALLOWED_BYTECODE_TYPES_MODE_SLOT").
+    bytes32 private constant ALLOWED_BYTECODE_TYPES_MODE_SLOT =
```

Comment:
IMO it is better to write to standard storage layout. Deprecation of variables isn't a concern. Bth my main concern would be making it simple and transparent with upgrades (all deprecated variables should be stored in the storage as a legacy, so no one ever will be confused and use the same storage slot for a different purpose)